src/main/java/com/hubspot/jinjava/interpret/LazyExpression.java

```diff
@@ -6,20 +6,26 @@
 public class LazyExpression implements Supplier {
   private final Supplier supplier;
   private final String image;
+  private final boolean memoize;
   private Object jsonValue = null;
 
-  private LazyExpression(Supplier supplier, String image) {
+  private LazyExpression(Supplier supplier, String image, boolean memoize) {
```

Comment:
http://www.nichesoftware.co.nz/2016/05/15/code-gardening-bools-are-evil.html

Reply:
LazyExpression.ofWithoutMemoization feels icky.

We could add a public method `withoutMemoization()` that would return a new LazyExpression, imitating a more typical builder pattern minus the `build()` call, but that also feels a little smelly to me. Do you have strong thoughts about a cleaner/more idiomatic solution?

Reply:
Use an enum in the class instead of the boolean. 

```
enum MEMOIZATION {
  ON,
  OFF
}
```

Reply:
Fixed in cefde706